block-producer/src/txqueue/mod.rs

```diff
@@ -0,0 +1,132 @@
+use std::{fmt::Debug, sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use tokio::{sync::RwLock, time};
+
+use crate::{batch_builder::BatchBuilder, SharedProvenTx, SharedRwVec};
+
+// TRANSACTION QUEUE
+// ================================================================================================
+
+#[derive(Debug)]
+pub enum VerifyTxError {}
+
+#[async_trait]
+pub trait TransactionVerifier: Send + Sync + 'static {
+    async fn verify_tx(
+        &self,
+        tx: SharedProvenTx,
+    ) -> Result<(), VerifyTxError>;
+}
+
+#[derive(Debug)]
+pub enum AddTransactionError {
+    VerificationFailed(VerifyTxError),
+}
+
+#[async_trait]
+pub trait TransactionQueue: Send + Sync + 'static {
+    async fn add_transaction(
+        &self,
+        tx: SharedProvenTx,
+    ) -> Result<(), AddTransactionError>;
+}
+
+// DEFAULT TRANSACTION QUEUE
+// ================================================================================================
+
+pub struct DefaultTransactionQueueOptions {
+    /// The frequency at which we try to send transaction groups
+    pub send_tx_groups_frequency: Duration,
```

Comment:
minor nit (feel free to ignore): maybe this should be "polling_frequency" or something like that as we may or may not send transactions at each poll.

Reply:
I renamed to `build_batch_frequency` with a docstring that explains we run the "attempt to build batch" operation at that frequency.

I don't really like "poll" terminology in this case, because IMO "polling" is more appropriate when you are checking in at some frequency if some event has occurred (and is also associated to be the opposite of being "event driven").